### PR TITLE
Add a Prisma tutorial

### PR DESCRIPTION
**Pull request by qianl15**
Documentation for using Prisma.

### CODE REVIEW

File: docs/tutorials/using-prisma.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+description: Learn how to use Prisma schema and perform transactional updates
+---
+
+## Prisma Overview
+[Prisma](https://www.prisma.io/) is a popular open-source TypeScript ORM.
+The main idea is to define your application data models (entities) in [Prisma Schema](https://www.prisma.io/docs/orm/prisma-schema/overview), and then use Prisma to automatically generate [schema migrations](https://www.prisma.io/docs/orm/prisma-migrate/getting-started) as well as the corresponding [Prisma Client](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/introduction) for you to manage your data.
+DBOS supports Prisma as an alternative to [Knex](https://knexjs.org/) and [TypeORM](https://typeorm.io) for transactional data management.
+
+## Usage
+DBOS supports essentially direct use of Prisma, but a few additional steps are necessary to generate schema migrations and the client, and use a transaction in the context of a workflow.
```
Comment (maxdml) [inline docs/tutorials/using-prisma.md:13]:
```suggestion
DBOS supports direct use of Prisma with only a few additional configuration steps to generate schema migrations and the Prisma client. The client can then be used in the context of a workflow.
```

File: docs/tutorials/using-prisma.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+npm run build
+npx dbos migrate
+npx dbos start
+```
+
+To see that it's working, visit this URL in your browser: [http://localhost:3000/greeting/dbos](http://localhost:3000/greeting/dbos).  You should get this message: `Greeting 1: Hello, dbos!` Each time you refresh the page, the counter should go up by one.
+
+### Setting Up Prisma Schema
+
+In DBOS, Prisma Schema by default is located at `./prisma/schema.prisma` and is defined in [the same way as any other Prisma project](https://www.prisma.io/docs/orm/prisma-schema/overview), for example:
```
Comment (maxdml) [inline docs/tutorials/using-prisma.md:37]:
```suggestion
In DBOS, Prisma Schema is located, by default, at `./prisma/schema.prisma` and is defined in [the same way as any other Prisma project](https://www.prisma.io/docs/orm/prisma-schema/overview), for example:
```

File: docs/tutorials/using-prisma.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  app_db_name: 'hello_prisma'
+  password: ${PGPASSWORD}
+  connectionTimeoutMillis: 3000
+  app_db_client: prisma
+  migrate:
+    - npx prisma migrate deploy
+```
+
+:::info
+Prisma doesn't [support migration rollback](https://github.com/prisma/prisma/discussions/4617) for successfully applied migrations like other ORMs.
```
Comment (maxdml) [inline docs/tutorials/using-prisma.md:126]:
```suggestion
Prisma doesn't [support schema rollbacks](https://github.com/prisma/prisma/discussions/4617) for successfully applied migrations like other ORMs.
```

File: docs/tutorials/using-prisma.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+}
+```
+
+Note that the schema file requires a `$DATABASE_URL` environment variable to connect to a database. To make it easy, we include in our template a script to automatically generate a `.env` file under the `./prisma` folder with the correct database URL constructed from your `dbos-config.yaml` file, please see the [document below](#configuring-prisma).
+
+
+### Schema Management
+
+In production scenarios or when using DBOS Cloud, we strongly recommend you manage your database schema using migrations.
+[Prisma Migrate](https://www.prisma.io/docs/orm/prisma-migrate/getting-started) provides plenty of support for generating and managing migrations.
```
Comment (maxdml) [inline docs/tutorials/using-prisma.md:62]:
```suggestion
Make sure to check [Prisma Migrate](https://www.prisma.io/docs/orm/prisma-migrate/getting-started) documentation for generating and managing Prisma migrations.
```

File: docs/tutorials/using-prisma.md (hunk 3 above)
Comment (qianl15) [reply docs/tutorials/using-prisma.md:126]: It's schema migration rollback.

File: docs/tutorials/using-prisma.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+npx dbos start
+```
+
+To see that it's working, visit this URL in your browser: [http://localhost:3000/greeting/dbos](http://localhost:3000/greeting/dbos).  You should get this message: `Greeting 1: Hello, dbos!` Each time you refresh the page, the counter should go up by one.
+
+### Setting Up Prisma Schema
+
+In DBOS, the Prisma Schema is defined in [the same way as any other Prisma project](https://www.prisma.io/docs/orm/prisma-schema/overview).
+In this tutorial we assume it is located at the default path (`prisma/schema.prisma`), for example:
+
```
Comment (chuck-dbos) [inline docs/tutorials/using-prisma.md:35]: Do we not support ```prisma ?

File: docs/tutorials/using-prisma.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  console.log("Wrote database URL to the prisma/.env file.");
+} catch (error) {
+  console.error("Error writing prisma/.env file:", error.message);
+}
+```
+
+You also need to generate a [Prisma Client](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/introduction) from your Prisma schema before compiling your application.
+Therefore, we recommend that you set the build script in your `package.json` file as follows:
+```js title="package.json"
+{
```
Comment (chuck-dbos) [inline docs/tutorials/using-prisma.md:147]:
```json ??

**Review by chuck-dbos — APPROVED** [review]
Didn't try to run, but looks good.  See comments.

File: docs/tutorials/using-prisma.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+To see that it's working, visit this URL in your browser: [http://localhost:3000/greeting/dbos](http://localhost:3000/greeting/dbos).  You should get this message: `Greeting 1: Hello, dbos!` Each time you refresh the page, the counter should go up by one.
+
+### Setting Up Prisma Schema
+
+In DBOS, the Prisma Schema is defined in [the same way as any other Prisma project](https://www.prisma.io/docs/orm/prisma-schema/overview).
+In this tutorial we assume it is located at the default path (`prisma/schema.prisma`), for example:
+
+```javascript title="prisma/schema.prisma"
```
Comment (chuck-dbos) [inline docs/tutorials/using-prisma.md:36]: Do we not support ```prisma ?

**Review by kraftp — APPROVED** [review]

File: docs/tutorials/using-prisma.md (hunk 6 above)
Comment (qianl15) [reply docs/tutorials/using-prisma.md:147]:
```json doesn't have syntax highlighting.

File: docs/tutorials/using-prisma.md (hunk 7 above)
Comment (qianl15) [reply docs/tutorials/using-prisma.md:36]: No syntax highlighting for Prisma so far (we use Docusaurus built-in list of supported languages)

**Pull request merged**